ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/utils/Types.kt

```diff
@@ -0,0 +1,35 @@
+/**
+ * File contains enums with minimum and maximum values of corresponding types
+ */
+
+package com.akuleshov7.ktoml.utils
+
+/**
+ * @property min
+ * @property max
+ */
+public enum class IntegerLimitsEnum(public val min: Long, public val max: Long) {
+    BYTE(Byte.MIN_VALUE.toLong(), Byte.MAX_VALUE.toLong()),
+    INT(Int.MIN_VALUE.toLong(), Int.MAX_VALUE.toLong()),
+    LONG(Long.MIN_VALUE, Long.MAX_VALUE),
+    SHORT(Short.MIN_VALUE.toLong(), Short.MAX_VALUE.toLong()),
+    ;
+    // Unsigned values are not supported now, and I think
+    // that will not be supported, because TOML spec says the following:
+    // Arbitrary 64-bit signed integers (from −2^63 to 2^63−1) should be accepted and handled losslessly.
```

Comment:
Would you mind elaborating on why unsigneds can't be supported?

I'd recommend making a new GitHub issue 'support unsigned ints' (I could make the issue if you like) and adding the link in the source code, then it's easier to track and follow progress.

Reply:
No, I don't mind, but the main problem is related to the kotlinx.serialization. I cannot find decodeUInt in their library...

Reply:
Yeah, it's a little bit odd how KxS handles unsigneds. 

KxS JSON handles them natively though https://github.com/Kotlin/kotlinx.serialization/blob/v1.4.1/docs/value-classes.md#unsigned-types-support-json-only

I suspect that they should 'just work' though, since they have serializers https://github.com/Kotlin/kotlinx.serialization/blob/v1.4.1/core/commonMain/src/kotlinx/serialization/internal/ValueClasses.kt

How KxS JSON handles them is

1. unsigned numbers are value classes, so their `SerialDescriptor`s are 'inline'
2. it keeps a list of the SerialDescriptors for all unsigned numbers
   https://github.com/Kotlin/kotlinx.serialization/blob/dc950f5098162a3f8dd742d04b95f9a0405470e1/formats/json/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonEncoder.kt#L15-L24
4. there's a specific function [`AbstractDecoder.decodeInline(...)`](https://github.com/Kotlin/kotlinx.serialization/blob/dc950f5098162a3f8dd742d04b95f9a0405470e1/core/commonMain/src/kotlinx/serialization/encoding/AbstractDecoder.kt#LL37C39-L37C39), that will handle the unsigned numbers
5. then there's a specific decoder for the unsigned types https://github.com/Kotlin/kotlinx.serialization/blob/d7e58c2eeb02e1da87ad36d41ef25523fafb6935/formats/json/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonDecoder.kt#L367-L379

Reply:
We will need to add it then also... Yes, let's support it then :)
Looks like a workaround in KxS, looks like they forgot about unsigned in the very beginning